exec/exec.go

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2024-Present Defense Unicorns
+
+// Package exec provides a wrapper around the os/exec package
+package exec
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"sync"
+)
+
+// Config is a struct for configuring the Cmd function.
+type Config struct {
+	Print          bool
+	Dir            string
+	Env            []string
+	CommandPrinter func(format string, a ...any)
+	Stdout         io.Writer
+	Stderr         io.Writer
+}
+
+// PrintCfg is a helper function for returning a Config struct with Print set to true.
+func PrintCfg() Config {
+	return Config{Print: true}
+}
+
+// Cmd executes a given command with given config.
+func Cmd(command string, args ...string) (string, string, error) {
+	return CmdWithContext(context.TODO(), Config{}, command, args...)
+}
+
+// CmdWithPrint executes a given command with given config and prints the command.
+func CmdWithPrint(command string, args ...string) error {
+	_, _, err := CmdWithContext(context.TODO(), PrintCfg(), command, args...)
+	return err
+}
+
+// CmdWithContext executes a given command with given config.
+func CmdWithContext(ctx context.Context, config Config, command string, args ...string) (string, string, error) {
```

Comment:
`PrintCfg` does not really need to exist, nor does `CmdWithPrint`

If a consumer wants to print, they can set `Print: true` on their `Config` struct.

There should only be 2 functions:

```
func Cmd(config Config, command string, args ...string) (string, string, error) {
	return CmdWithContext(context.TODO(), config, command, args...)
}

func CmdWithContext(ctx context.Context, config Config, command string, args ...string) (string, string, error) {
...
```